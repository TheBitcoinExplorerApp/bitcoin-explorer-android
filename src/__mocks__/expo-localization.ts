const getLocales = () => [
  { country: "pt", languageTag: "pt-BR", languageCode: "pt", isRTL: false },
];

export default getLocales;
