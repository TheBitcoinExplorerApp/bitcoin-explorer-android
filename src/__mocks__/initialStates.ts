import { BlockType } from "src/components/BlocksInfo/components/Block/type";
import { AddressInfoType, TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks: BlockType[] = [
  {
    blockHash: "",
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,

    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};

export const initialStateFormattedTransactions = [
  {
    transactionId: "0",
    value: 0,
    fee: 0,
  },
  {
    transactionId: "1",
    value: 0,
    fee: 0,
  },
];

export const initialStateAddress: AddressInfoType = {
  address: "",
  balance: 0,
  totalAmountReceived: 0,
  totalAmountSent: 0,
};
