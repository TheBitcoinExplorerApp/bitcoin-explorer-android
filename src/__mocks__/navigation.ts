const mockNavigation = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNavigation = jest.requireActual("@react-navigation/native");
  return {
    ...actualNavigation,
    useNavigation: () => ({
      navigate: mockNavigation,
    }),
  };
});
