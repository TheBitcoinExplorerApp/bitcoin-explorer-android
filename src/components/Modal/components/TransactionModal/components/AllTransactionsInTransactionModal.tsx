/* eslint-disable camelcase */
/* eslint-disable no-unsafe-optional-chaining */
/* eslint-disable import/no-extraneous-dependencies */
import { nanoid } from "nanoid";
import { View, Text, StyleSheet } from "react-native";
import { TransactionType } from "src/components/Modal/types";
import useAppDataStore from "src/context/DataProvider";

type AllTransactionsInTransactionModalProps = {
  inputTransactions: TransactionType["inputTransactions"];
  outputTransactions: TransactionType["outputTransactions"];
};

const styles = StyleSheet.create({
  title: {
    color: "#FFF",
    fontSize: 15,
    fontWeight: "600",
  },
  container: {
    gap: 16,
    height: "100%",
    backgroundColor: "#101427",
  },
  transactionContainer: {
    backgroundColor: "#1d2133",
    borderRadius: 8,
    paddingVertical: 5,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    maxWidth: "100%",
  },
  inputTransactionsContainer: {
    gap: 5,
    alignItems: "center",
  },
  outputTransactionsContainer: {
    gap: 5,
    alignItems: "center",
  },
  eachTransactionOutputContainer: {
    width: "100%",
    alignItems: "center",
  },
  transactionContentContainer: {
    alignItems: "center",
    width: "100%",
  },
  transactionAddress: {
    color: "#DF7800",
    fontSize: 16,
  },
  text: {
    color: "#FFF",
    fontSize: 13,
    textAlign: "right",
  },
  destinationTransactionSeparator: {
    color: "#D9D9D9",
    fontSize: 28,
    fontWeight: "800",
  },
});

// #TODO: VERIFY THE TYPES TO CONFIRM IF CAN HAPPEN NULL INCOME

export default function AllTransactionsInTransactionModal(
  props: AllTransactionsInTransactionModalProps,
) {
  const { inputTransactions, outputTransactions } = props;
  const { i18n } = useAppDataStore();

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{i18n.t("inputs_and_outputs")}</Text>

      <View style={styles.transactionContainer}>
        <View style={styles.inputTransactionsContainer}>
          {inputTransactions.map((inputTransaction) => {
            const { prevout } = inputTransaction;

            return (
              <View
                style={styles.transactionContentContainer}
                key={`${nanoid()}`}
              >
                <Text style={styles.transactionAddress}>
                  {(prevout?.scriptpubkey_address)
                    .toString()
                    .slice(0, 14)
                    .concat("...")}
                </Text>

                <Text style={styles.text}>
                  {(prevout?.value / 100000000).toString()} BTC
                </Text>
              </View>
            );
          })}
        </View>

        <Text style={styles.destinationTransactionSeparator}>{">"}</Text>

        <View style={styles.outputTransactionsContainer}>
          {outputTransactions.map((outputTransaction) => {
            const { value, scriptpubkey_address } = outputTransaction; // #TODO: outputTransactions CHANGE TYPE TO CAMEL CASE

            return (
              <View
                key={`${nanoid()}`}
                style={styles.eachTransactionOutputContainer}
              >
                <Text style={styles.transactionAddress}>
                  {scriptpubkey_address.toString().slice(0, 14).concat("...")}
                </Text>
                <Text style={styles.text}>{value / 100000000} BTC</Text>
              </View>
            );
          })}
        </View>
      </View>
    </View>
  );
}
