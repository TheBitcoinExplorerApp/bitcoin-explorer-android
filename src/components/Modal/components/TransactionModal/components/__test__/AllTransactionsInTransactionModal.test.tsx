/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import { initialStateTransaction } from "src/__mocks__/initialStates";
import AllTransactionsInTransactionModal from "../AllTransactionsInTransactionModal";

describe("AllTransactionsInTransactionModal", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <AllTransactionsInTransactionModal
          inputTransactions={initialStateTransaction.inputTransactions}
          outputTransactions={initialStateTransaction.outputTransactions}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
