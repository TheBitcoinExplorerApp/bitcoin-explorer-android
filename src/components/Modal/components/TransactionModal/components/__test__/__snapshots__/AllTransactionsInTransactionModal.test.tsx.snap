// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllTransactionsInTransactionModal renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#101427",
      "gap": 16,
      "height": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    Entradas e Sa√≠das
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1d2133",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "maxWidth": "100%",
        "paddingVertical": 5,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "gap": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 16,
            }
          }
        >
           ...
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 13,
              "textAlign": "right",
            }
          }
        >
          0
           BTC
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#D9D9D9",
          "fontSize": 28,
          "fontWeight": "800",
        }
      }
    >
      &gt;
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 16,
            }
          }
        >
           ...
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 13,
              "textAlign": "right",
            }
          }
        >
          0
           BTC
        </Text>
      </View>
    </View>
  </View>
</View>
`;
